src/main/java/org/symphonyoss/symphony/messageml/elements/Element.java

```diff
@@ -373,7 +373,7 @@ void assertContentModel(Collection<Class<? extends Element>> permittedChildren)
    * Check that the element's allowed parents are limited to the specified element types.
    */
   void assertParent(Collection<Class<? extends Element>> permittedParents) throws InvalidInputException {
-    if (!permittedParents.contains(this.getParent().getClass())) {
+    if (!assertAboveParents(this, permittedParents)) {
```

Comment:
You are with the correct idea here, but we have to be cautious over this as this method is used by other elements not related to forms or SymphonyElements themselves. So I would suggest you to have both methods, one "assertParent" and one "assertParentAtAnyLevel" or something similar. WDYT?